### DISCUSSION TITLE
Suggested approach for automatically namespacing keys

### DISCUSSION THREAD

**Discussion by killergerbah** [xetorthio/jedis#3765] [Ideas]
On our project it is useful to be able to namespace keys such that applications belonging to different environments automatically read and write from distinct key spaces. For example, prefix keys belonging to the `dev1` environment with `dev1:`, from the `dev2` environment with `dev2:` and so on. We had previously achieved this by implementing wrappers around each Redis command method, but this results in extremely large interfaces that are difficult to maintain.

It seems to me that the best way to do this in Jedis 5.x would be to customize the `CommandObjects` being used by `UnifiedJedis`. However, the constructor that would allow this type of customization is currently private.

I have two questions:
1. Would this be sufficient motivation to make `UnifiedJedis` more customizable by making some of the constructors either `protected` or `public`?
2. Are there other straightforward ways to implement namespaced keys that I am not thinking of?

**Comment by sazzad16** [comment]
> 1. Would this be sufficient motivation to make `UnifiedJedis` more customizable by making some of the constructors either `protected` or `public`?

You can submit a simple pull request as a proof of concept. Then it'll be easier for further discussion.

**Reply by killergerbah** [reply]
Thank you for your reply. I have created a proof of concept here: https://github.com/redis/jedis/pull/3770

**Comment by sazzad16** [comment]
> 2. Are there other straightforward ways to implement namespaced keys that I am not thinking of?

I don't know any **straightforward** way.
